src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -284,13 +287,30 @@ private void ChangeProjectProperty<T>(
 
                     if (!isFullyLoaded)
                     {
-                        TryReportCompilationThrownAway(_projectSystemProjectFactory.Workspace.CurrentSolution, Id);
+                        var reportCompilationThrownAway = true;
+
+                        // Preprocessor directive only ParseOptions changes have special handling in DocumentState.UpdateParseOptionsAndSourceCodeKind.
+                        // We don't want to report telemetry for those changes or for the initial evaluation result.
+                        if (newValue is ParseOptions newParseOption)
+                        {
+                            if (oldValue is ParseOptions oldParseOptions)
+                            {
+                                var syntaxTreeFactoryService = _projectSystemProjectFactory.SolutionServices.GetRequiredLanguageService<ISyntaxTreeFactoryService>(Language);
+
+                                if (syntaxTreeFactoryService.OptionsDifferOnlyByPreprocessorDirectives(oldParseOptions, newParseOption))
+                                    reportCompilationThrownAway = false;
+                            }
+                            else
+                            {
+                                reportCompilationThrownAway = false;
+                            }
+                        }
+
+                        if (reportCompilationThrownAway)
+                            TryReportCompilationThrownAway(_projectSystemProjectFactory.Workspace.CurrentSolution, Id);
```

Comment:
mentioned offline.  but there just isn't that many reasons why a prse option could change.  we should actually detect what actually changed and report that as part of telem.  For example, i'd love to know thigns like "90% of throwaways were caused by LangVersion changing", or something like that.